### DISCUSSION TITLE
Realtime Subscription Scalability?

### DISCUSSION THREAD

**Discussion by masnwilliams** [supabase/supabase#8095] [Questions] [answered]
Hello! I was wondering how scalable realtime subscriptions are.

I have a marketplace and was wondering if we could subscribe to the product row for each product page. This would be on a large scale of tens of thousands of product pages so I am wondering if it would eat up a ton of bandwidth or any other considerations we would have to take.

Thanks!

**Comment by w3b6x9** [comment, marked as answer]
@johnmasonwilliams will a client be accessing a single product page at a time or all product pages at the same time? You should be fine as long as a client is subscribing to a page, and then when navigating to another page, have that subscription removed and open another one on the new product page.

**Reply by masnwilliams** [reply]
The user would be accessing a single product page at a time so it would be possible to subscribe on load and then unsubscribe either on page change or if they unfocus from that tab.

I think it would be a cool way to experience a website and actually see other users there instead of just being alone.

**Reply by Ziinc** [reply]
You would only need to worry about scalability when there are many tens of thousands of users on a single page at a single point in time, and if you are broadcasting table changes to all of them at the same time.
